hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineStateManager.java

```diff
@@ -65,8 +65,7 @@ void updatePipelineState(HddsProtos.PipelineID pipelineIDProto,
 
   void addContainerToPipeline(
       PipelineID pipelineID,
-      ContainerID containerID
-  ) throws IOException;
+      ContainerID containerID, boolean checkPipelineClosed) throws IOException;
```

Comment:
Instead of changing the interface, how about adding it as a member variable, set via `PipelineStateManagerImpl.Builder` here:

https://github.com/apache/ozone/blob/086e303f145e1f7a4ded5f403e0255c898fef721/hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/ReconPipelineManager.java#L78-L84